backstory/src/main/java/com/google/sps/servlets/AnalyzeImageServlet.java

```diff
@@ -0,0 +1,265 @@
+      BlobstoreManager blobstoreManager = blobstoreManagerFactory.newInstance();
+      final String blobKeyString =
+          blobstoreManager.getUploadedFileBlobKeyString(request, "image-upload");
+      // The raw byte array representing the image will be used for image analytics.
+      final byte[] bytes = blobstoreManager.getBlobBytes(request, "image-upload");
+
+      // Validate that an image was actually uploaded.
+      if (bytes == null || blobKeyString == null) {
+        // Redirect back to the HTML page.
+        response.sendError(400, "Please upload a valid image.");
+
+      } else {
+        // Generate a list of AnnotatedImages, with each annotatedImage consisting of an image with
+        // labels.
+        ImagesManager imagesManager = imagesManagerFactory.newInstance();
+        List<byte[]> imagesAsByteArrays = new ArrayList<>();
+        imagesAsByteArrays.add(bytes);
+        List<AnnotatedImage> annotatedImages =
+            imagesManager.createAnnotatedImagesFromImagesAsByteArrays(imagesAsByteArrays);
+        // Currently, Backstory only supports single image uploads.
+        // which is why we only get the first annotatedImage element here from annotatedImages.
+        AnnotatedImage annotatedImage = annotatedImages.get(0);
+        List<String> descriptions = annotatedImage.getLabelDescriptions();
+
+        PromptManager promptManager = new PromptManager(descriptions);
+        // The delimiter for the MVP prompt will be tentatively be " and ", ex:
+        // "<label1> and <label2> and <label3>"
+        String prompt = promptManager.generatePrompt(" and ");
+
+        // Tentative backstory generation parameters: a 200 word-long story, with a .7 temperature.
+        StoryManager storyManager = storyManagerFactory.newInstance(prompt, 200, .7);
+
+        // We loop until a story succesfully generates, this is necessary because of an unavoidable
+        // memory leak in the GPT2 container which causes generation to occasionaly fail.
+        String rawBackstory = "";
+        Boolean generateTextFailed = true;
+        while (generateTextFailed) {
+          try {
+            rawBackstory = storyManager.generateText();
+            generateTextFailed = false;
+          } catch (RuntimeException exception) {
+            System.err.println(exception);
+            generateTextFailed = true;
+          }
+        }
+
+        String backstory = "";
+        try {
+          StoryAnalysisManager storyAnalysisManager = storyAnalysisManagerFactory.newInstance();
+          StoryDecision storyDecision = storyAnalysisManager.generateDecision(rawBackstory);
+          backstory = storyDecision.getStory();
+        } catch (NoAppropriateStoryException | APINotAvailableException exception) {
+          response.sendError(400,
+              "Sorry! No appropriate Backstory was found for your image. Please try again with another image.");
```

Comment:
Should there be a `return` here? Without it it looks like you'd end up persisting a "" story below.

Reply:
Absolutely! I'll add it.